stdlib/AllocProfile/src/AllocProfile.jl

```diff
@@ -0,0 +1,189 @@
+module AllocProfile
+
+using Base.StackTraces: StackTrace, StackFrame, lookup
+using Base: InterpreterIP, CodeInfo, _reformat_bt
+
+const ExtendedEntryObj = Union{
+    Base.CodeInfo,
+    Module,
+    Base.InterpreterIP,
+    Core.Compiler.InterpreterIP
+}
+
+# matches RawAllocResults on the C side
+struct RawAllocProfile
+    alloc_types::Vector{Ptr{Type}}
+    alloc_sizes::Vector{Csize_t}
+    alloc_bts::Vector{Vector{Ptr{Cvoid}}}
+    alloc_bt2s::Vector{Vector{ExtendedEntryObj}}
+
+    # sampling parameter
+    skip_every::Csize_t
+    # sampling state
+    alloc_counter::Csize_t
+    last_recorded_alloc::Csize_t
+    
+    # TODO: get frees (garbage profiling) working again
+    # frees_by_type::Dict{Type,UInt}
+    # type_address_by_value_address::Dict{}
+
+    function RawAllocProfile(skip_every::Int)
+        return new(
+            Vector{Type}(),
+            Vector{Csize_t}(),
+            Vector{Vector{Ptr{Cvoid}}}(),
+            Vector{Vector{ExtendedEntryObj}}(),
+            skip_every,
+            0,
+            0
+        )
+    end
+end
+
+# default implementation actually gives a fatal type inference
+# error while printing. lol
+function Base.show(io::IO, ::RawAllocProfile)
+    print(io, "RawAllocProfile")
+end
+
+# pass this in, push to it
+const g_profile = Ref{RawAllocProfile}()
+
+function start(skip_every::Int=0)
+    g_profile[] = RawAllocProfile(skip_every)
+    ccall(
+        :jl_start_alloc_profile,
+        Cvoid,
+        (Cint, Ref{RawAllocProfile}),
+        skip_every,
+        g_profile
+    )
+end
+
+function stop()
+    ccall(:jl_stop_alloc_profile, Cvoid, ())
+end
+
+# decoded results
+
+struct Alloc
+    type::Type
+    stacktrace::StackTrace
+    bytes_allocated::Int
+end
+
+struct AllocResults
+    allocs::Vector{Alloc}
+    frees::Dict{Type,UInt}
+    bt2_length::Int
+end
+
+function Base.show(io::IO, ::AllocResults)
+    print(io, "AllocResults")
+end
+
+const BacktraceEntry = Union{Ptr{Cvoid}, InterpreterIP}
+const BacktraceCache = Dict{BacktraceEntry,Vector{StackFrame}}
+
+# loading anything below this seems to segfault
+# TODO: use filter out special types before we get here
+TYPE_PTR_THRESHOLD = 0x0000000100000000
+
+function load_type(ptr::Ptr{Type})::Type
+    if UInt(ptr) < TYPE_PTR_THRESHOLD
+        return Missing
+    end
+    return unsafe_pointer_to_objref(ptr)
+end
+
+# same as _reformat_bt, except has a more specific type for bt2
```

Comment:
this makes it avoid an illegal instruction error:

```
signal (4): Illegal instruction: 4
in expression starting at REPL[4]:1
decode at /Users/vilterp/code/julia/stdlib/AllocProfile/src/AllocProfile.jl:114
unknown function (ip: 0x149eba06e)
_jl_invoke at /Users/vilterp/code/julia/src/gf.c:0 [inlined]
ijl_apply_generic at /Users/vilterp/code/julia/src/gf.c:2451
jl_apply at /Users/vilterp/code/julia/src/./julia.h:1773 [inlined]
do_call at /Users/vilterp/code/julia/src/interpreter.c:126
eval_body at /Users/vilterp/code/julia/src/interpreter.c:467
```

not sure why.

Tried making the same edit in the stdlib, but it caused a method error while building the stdlib. Will try again tomorrow to find a type that works for both.